pallets/pallet-bonded-coins/src/tests/curves/lmsr.rs

```diff
@@ -0,0 +1,133 @@
+use std::str::FromStr;
+
+use crate::{
+	curves::{BondingFunction, LMSRCalculation},
+	mock::{assert_relative_eq, Float},
+};
+
+#[test]
+fn mint_first_coin() {
+	// Create curve with liquidity parameter b=100_000_000, and passive issuance=0
+	let m = Float::from_num(100_000_000);
+	let passive_issuance = Float::from_num(0);
+
+	let curve = LMSRCalculation { m, passive_issuance };
+
+	let low = Float::from_num(0);
+	let high = Float::from_num(1);
+
+	// Costs for existing supply:  100000000 * ln(e^(0/100000000) + e^(0/100000000)) = 69314718.055994530942
+	// Costs for new supply: 100000000 * ln(e^(1/100000000) + e^(0/100000000)) = 69314718.555994532192
+	// Costs to mint the first coin: 69314718.555994532192 - 69314718.055994530942 = 0.50000000124972321215 -> 0.50000000124972321215
+	let costs = curve.calculate_costs(low, high).unwrap();
+
+	let expected_costs = Float::from_str("0.50000000124972321215").unwrap();
+
+	assert_relative_eq(costs, expected_costs, Float::from_str("0.00000002").unwrap());
```

Comment:
Is `00000002` your defined precision?

Reply:
Could this be a parameter of the curve itself, and called internally in case this assertion fails?

Reply:
The error is different depending on the input. I assume we can have a min required precision

Reply:
But the value would be pretty arbitrary.

Reply:
It seems a bit counter-intuitive that there's no minimum precision, and that it depends on the operation. I still think it would benefit to have it as part of the curve, and fail operations that result in less precision than configured. But again, this is out of my knowledge area, so do of this comment as you please.

Reply:
Rounding issues can not be completely resolved. we have to accept them. I also can't guarantee a minimum level of precision, as it depends heavily on the inputs, selected coefficients, operations, and the chosen fixed-point type. What we can guarantee is that the error is reproducible and independent of the FPU. The calculated exchange rate, even if mathematically inaccurate, is still the authoritative exchange rate